src/app/vault/vault.component.ts

```diff
@@ -547,56 +534,73 @@ export class VaultComponent implements OnInit, OnDestroy {
     this.go();
   }
 
-  async clearGroupingFilters() {
-    this.searchBarService.setPlaceholderText(this.i18nService.t("searchVault"));
-    await this.ciphersComponent.reload();
-    this.clearFilters();
-    this.go();
-  }
-
-  async filterFavorites() {
-    this.searchBarService.setPlaceholderText(this.i18nService.t("searchFavorites"));
-    await this.ciphersComponent.reload((c) => c.favorite);
-    this.clearFilters();
-    this.favorites = true;
-    this.go();
-  }
-
-  async filterDeleted() {
-    this.searchBarService.setPlaceholderText(this.i18nService.t("searchTrash"));
-    this.ciphersComponent.deleted = true;
-    await this.ciphersComponent.reload(null, true);
-    this.clearFilters();
-    this.deleted = true;
+  async applyVaultFilter(vaultFilter: VaultFilter) {
+    this.searchBarService.setPlaceholderText(
+      this.i18nService.t(this.calculateSearchBarLocalizationString(vaultFilter))
+    );
+    this.activeFilter = vaultFilter;
+    await this.ciphersComponent.reload(this.buildFilter());
     this.go();
   }
 
-  async filterCipherType(type: CipherType) {
-    this.searchBarService.setPlaceholderText(this.i18nService.t("searchType"));
-    await this.ciphersComponent.reload((c) => c.type === type);
-    this.clearFilters();
-    this.type = type;
-    this.go();
-  }
+  private calculateSearchBarLocalizationString(vaultFilter: VaultFilter): string {
```

Comment:
Should this be a public method on the filter class? It doesn't refer to any methods/properties on this class.